tests/test_qubit_device.py

```diff
@@ -764,6 +765,26 @@ def circuit_3(x, y):
             node_3(0.432, 0.12)
         assert dev_1.num_executions == num_evals_1 + num_evals_3
 
+    def test_different_hash(self):
+        """Test that executing different tapes affects the circuit hash."""
+        dev = qml.device("default.qubit", wires=2)
+
+        with QuantumTape() as tape0:
+            qml.RZ(0.3, wires=[0])
+            qml.expval(qml.PauliX(0))
+
+        tape0.execute(dev)
+        orig_hash = dev.circuit_hash
+
+        with QuantumTape() as tape1:
+            qml.RY(1.3, wires=[0])
+            qml.RX(0.9, wires=[0])
+            qml.expval(qml.PauliX(0))
+
+        tape1.execute(dev)
+        new_hash = dev.circuit_hash
+        assert orig_hash != new_hash
```

Comment:
Nice! Is there a test already that checks that executing the same tape twice results in the same hash?

Reply:
Oh good point! Just added one.